kernel/reduction.ml

```diff
@@ -938,6 +938,18 @@ let hnf_prod_app env t n =
 let hnf_prod_applist env t nl =
   List.fold_left (hnf_prod_app env) t nl
 
+let hnf_prod_applist_assum env n c l =
```

Comment:
Why do we take `n`? Is it just a safety measure?

Reply:
Oh, no, I understand now, it is to make sure that all local definitions are unfolded, if they are included in `n`.

Reply:
Yes, exactly!